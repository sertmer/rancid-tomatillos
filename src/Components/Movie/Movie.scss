* {
  border: 1px solid red;
}

.movie-card {
  border: 1px solid black;
  height: 80vh;
  width: 50vh;
}

.movie-poster-img {
  height: 75%;
  width: 100%;
}

.movie-card-bottom {
  background: rgb(133,143,143);
  background: linear-gradient(0deg, rgba(133,143,143,1) 0%, rgba(10,7,0,1) 100%);
  text-align: center;
}

.movie-title {
  color: white;
  text-align: center;
}

.average-rating {
  color: #FFAC33;
}

.movie-button {
  background-color: #30BD9E;
  border-radius: 20px;
  font-size: 1.25em;
  margin-bottom: .75em;
    &:hover {
    font-weight: bold;
    cursor: pointer;
  }
}
