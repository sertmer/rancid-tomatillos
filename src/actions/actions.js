export const getMovies = movies => ({
  type: 'GET_MOVIES',
  movies
});

export const currentUser = user => ({
  type: 'SET_CURRENT_USER',
  name: user.user.name,
  id: user.user.id,
  ratings: []
});

export const loggedIn = loggedIn => ({
  type: 'LOG_IN',
  loggedIn
});

export const getCurrentMovie = movie => ({
  type: 'GET_MOVIE',
  movie
});

export const getRatings = ratings => ({
  type: 'GET_RATINGS',
  ratings
});

