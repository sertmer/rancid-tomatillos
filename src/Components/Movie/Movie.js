import React from 'react';
import './Movie.scss';
import { Link } from 'react-router-dom';
import { getCurrentMovie } from '../../actions/actions';
import { connect } from 'react-redux';

export const Movie = (props) => {
  return(
    <section className="movie-card" onClick={() => props.getCurrentMovie(props.movie)}>
      <Link to={`/movies/${props.movie.id}`}>
      <img className='movie-poster-img' src={props.movie.poster_path}/>
      </Link>
      <div className="movie-card-bottom">
        <p className="average-rating">
        Avg. Rating: {Math.round(props.movie.average_rating)} / 10
        </p>
        <h2 className="movie-title">{props.movie.title}</h2>
        <Link to={`/movies/${props.movie.id}`}>
          <button className="movie-button" onClick={() => props.getCurrentMovie(props.movie)}>START THE SHOW</button>
        </Link>
      </div>
    </section>
  )
}

export const mapDispatch = dispatch => ({
  getCurrentMovie: movie => dispatch(getCurrentMovie(movie))
})

export default connect(null, mapDispatch)(Movie)
