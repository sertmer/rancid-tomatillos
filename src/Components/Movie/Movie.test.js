import React from 'react';
import { Movie, mapDispatch } from './Movie';
import { shallow } from 'enzyme';
import { getCurrentMovie } from '../../actions/actions';


describe('Movie', () => {
  let mockProps = {
    movie: {
      id: 1,
      title: 'Erties Dirties',
      average_rating: 2,
      poster_path: 'www.weewoo.com'
    },
    type: 'GET_MOVIE'
  }

  it.skip('should match the snapshot', () => {
    let wrapper = shallow(<Movie
      movie={mockProps.movie}
      type={mockProps.type}
    />)

    expect(wrapper).toMatchSnapshot()
  });

describe('mapDispatch', () => {
  it('should call dispatch with a getCurrentMovie action when getCurrentMovie is called from props', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = getCurrentMovie(mockProps);
    const mappedProps = mapDispatch(mockDispatch);

    mappedProps.getCurrentMovie(mockProps);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
})
})
