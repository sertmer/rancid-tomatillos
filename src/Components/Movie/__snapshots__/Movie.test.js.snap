// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot 1`] = `
"<section className=\\"movie-card\\" onClick={[Function: onClick]}>
  <Link to=\\"/movies/1\\">
    <img className=\\"movie-poster-img\\" src=\\"www.weewoo.com\\" />
  </Link>
  <div className=\\"movie-card-bottom\\">
    <p className=\\"average-rating\\">
      Avg. Rating: 
      2
       / 10
    </p>
    <h2 className=\\"movie-title\\">
      Erties Dirties
    </h2>
    <Link to=\\"/movies/1\\">
      <button className=\\"movie-button\\" onClick={[Function: onClick]}>
        START THE SHOW
      </button>
    </Link>
  </div>
</section>"
`;
